### DISCUSSION TITLE
关于SQL的一些想法

### DISCUSSION THREAD

**Discussion by WT1W** [liusy58/NJU_AdvancedProgramming_SP22#38] [General]
请问我可不可以定义卖家，买家，管理员三个类，在这三个类里面都会有一个SQL的生成函数对命令进行生成，再分别在三个类里用不同的解析函数进行解析，也就是说把SQL分成独立的三块，不提供统一的接口。还是说这个项目要求把SQL放到一块里面实现？

**Comment by binbin00** [comment]
对实现的方式不做要求，保证指令日志文件中的指令都正确即可。
对类SQL指令部分不会单独测试，只会查看指令日志文件和检查合成与解析指令的源代码。

**Reply by bliss-bug** [reply]
请问如果比如卖家发布商品时输入商品的属性是非法的，还需要在日志中显示SQL指令吗？

**Reply by RicoloveFeng** [reply]
> 请问如果比如卖家发布商品时输入商品的属性是非法的，还需要在日志中显示SQL指令吗？

应当在输入无效时就停止处理并报错

**Reply by bliss-bug** [reply]
好的，谢谢
